crates/provider-sdk/src/provider.rs

```diff
@@ -484,16 +482,16 @@ async fn delete_link_for_provider<P>(
 where
     P: Provider,
 {
-    if ld.source_id == connection.provider_key {
+    if ld.source_id == connection.provider_id {
         if let Err(e) = provider.delete_link_as_source(&ld.target).await {
-            error!(error = %e, "failed to delete link");
+            error!(error = %e, target = &ld.target, "failed to delete link to component");
         }
-    } else if ld.target == connection.provider_key {
-        connection.delete_link(&ld.source_id).await;
+    } else if ld.target == connection.provider_id {
         if let Err(e) = provider.delete_link_as_target(&ld.source_id).await {
-            error!(error = %e, "failed to delete link");
+            error!(error = %e, source = &ld.source_id, "failed to delete link from component");
```

Comment:
It's a bit weird since `source`s are almost always components (for now), but I've been calling this `component_id = source_id` to match the content of the error... 

Maybe someday we'll go through and be more consistently generic about what "source" means, but IMO if we're calling it a "component" then we should have `component_id` there

Reply:
resolving for 🚀